V2.0.0

mssalvatore opened the pull request:
Not all projects that use ServiceKit use it's API-related features. For example, some projects rely on Service-Kit solely for it's logging. It's therefore unnecessary and cumbersome for FastAPI and its sub-dependencies to be installed.

The main goal of this release is to only install API related dependencies if the [api] extra is installed.

```
# Changelog

## [2.0.0] - 2025-11-13
### Added
- `service_kit.logging.intercept_uvicorn_loggers()`

### Changed
- FastAPI, Uvicorn, and related dependencies are only installed if the \[api\]
  extra is specified.
- RequestIDMiddleware to use UUIDv7 instead of ULID if UUIDv7 is available.
```

mssalvatore committed:
Add [api] extras

mssalvatore committed:
Fix formatting in service_kit/types.py

mssalvatore committed:
Add --all-extras flag to developer installation instructions

mssalvatore committed:
Add instructions to install Service-Kit with extras in README.md

mssalvatore committed:
Add version specifier for psycopg dependency

mssalvatore committed:
Add service_kit.logging.intercept_uvicorn_loggers()

Adds a new function that handles configuring uvicorn's loggers. The new
function is aware that the uvicorn may not be available, since it's only
installed with the [api] extra.

mssalvatore committed:
Add changelog entry for the new [api] package extra

mssalvatore committed:
Use UUIDv7 in RequestIDMiddleware if it's available

mssalvatore committed:
Bump version to v2.0.0

mssalvatore committed:
Use contextlib.suppress() instead of try/except

ilija-lazoroski committed:
Added `service_kit.utils.Timer`

cakekoa commented:
I see the following when trying to use the test logger without having the `api` extra:

```
ImportError while loading conftest '/Users/kkaaikal/source/ocp-suggestions/tests/conftest.py'.
tests/conftest.py:2: in <module>
    from service_kit.testing import configure_test_logger  # noqa: F401
.venv/lib/python3.13/site-packages/service_kit/testing/__init__.py:1: in <module>
    from .api import request_id
.venv/lib/python3.13/site-packages/service_kit/testing/api.py:5: in <module>
    from service_kit.api import RequestID
.venv/lib/python3.13/site-packages/service_kit/api/__init__.py:16: in <module>
    require_extra("api", ["fastapi", "ulid", "uvicorn"])
.venv/lib/python3.13/site-packages/service_kit/api/__init__.py:9: in require_extra
    raise ImportError(
E   ImportError: The 'api' extra is required in order to use 'service_kit.api'. Install with:
E
E       pip install service-kit[api]
E
E   Missing: fastapi, ulid, uvicorn
```

mssalvatore committed:
Remove unnecessary `self._elapsed` attribute from Timer

mssalvatore committed:
Merge branch 'add-timer-to-utils' into v2.0.0

PR #9

mssalvatore committed:
Split service_kit.testing

The test utilities associated with the API features only need to be
available if the `api` extra is installed. Breaking up
`service_kit.testing` is the simplest way to ensure that test utilities
associated with specific submodules are only installed if the related
extras are specified.

mssalvatore committed:
Adjust v2.0.0 release date
